### PR TITLE
Move all the configuration from setup.py into setup.cfg

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
+[options]
+python_requires = >=3.6
+packages = find:
+include_package_data = True
+install_requires =
+  numpy>=1.11
+  scipy
+  matplotlib>=1.3
+  pandas>=0.12
+  astropy>=3.1
```
Comment (nabobalis) [inline setup.cfg]: Should we look at updating these numbers for 1.0?

File: ah_bootstrap.py [hunk 2]
```diff
@@ -45,18 +45,99 @@
...
+
+    # We want the Python version as a string, which we can get from the platform module
+    import platform
+    python_version = platform.python_version()
+
+    if not req.specifier.contains(python_version):
+        if parent_package is None:
+            print("ERROR: Python {} is required by this package".format(req.specifier))
+        else:
+            print("ERROR: Python {} is required by {}".format(req.specifier, parent_package))
```
Comment (dpshelio) [inline ah_bootstrap.py]: why not to `raise` instead of `print`s??

File: setup.cfg [hunk 3]
```diff
@@ -1,27 +1,61 @@
 [metadata]
-package_name = sunpy
+name = sunpy
+package_name = sunpy # astropy compat
 description = SunPy: Python for Solar Physics
-long_descripton = An open-source Python library for Solar Physics data analysis and visualization.
+long_descripton = file: README.rst
```
Comment (dpshelio) [inline setup.cfg]: wow! I didn't know this!

**Comment by nabobalis** [comment]
Just add the dev category and I think we can merge this. 

Tho the 32bit build seems to be a real failure, the others are just known issues.

**Comment by nabobalis** [comment]
Test failures unrelated to PR.

**Pull request merged**